### DISCUSSION TITLE
Connecting multiple data centres together

### DISCUSSION THREAD

**Discussion by ulfaric** [manoelcampos/cloudsim-plus#390] [Q&A] [answered]
Hi,

I could not find examples that show how to use root switch and aggregate switch, for connecting data centers together. Could anyone share some insights?

**Comment by jlxcsong** [comment]
您好！您发的邮件我已收到！

**Comment by manoelcampos** [comment, marked as answer]
Hello @ulfaric 

The network module is inherited from CloudSim and it lacks some features. I've just done some refactoring and design improvements but it actually lacks a way to connect root and aggregate switches.

I've created issue #391 to check that.

**Comment by ulfaric** [comment]
Thanks for the clarification. :)

I assume for the edge switches, they are working properly?

Best regards

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by manoelcampos** [reply]
Yes, they are.

**Comment by ulfaric** [comment]
Hi,

I just come up another question. :)

If all hosts have been set with a bandwidth A, then they are connected to a edge switch with downlink bandwidth B. Assume A > B, then the hosts bandwidth will be limited to B?

Best regards,
Yifei

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by manoelcampos** [reply]
Host bandwidth and switches bandwidth is another thing which is very confusing. I fixed some issues and improved documentation, but there some work to be done yet.
Host bandwidth is used just to check if the bandwidth required by VMs is enough when placing those VMs. That bw is not even considered when actually sending or receiving packets. You can check that all examples assing bw for hosts, but just the network examples in fact send and receive packets, actually performing network communication.

We could think of the Host bandwidth just as a way to set traffic shapping, i.e, to ensure a Host won't use more than a given amount of bw. If you are creating a network example using switches and actually transmitting packets, the switches bw will be considered. If the host try to use more bw than the switch capacity (which is you case), there will be traffic congestion and delay. You could even see some logs indicating that Cloudlet processing was delayed due to network congestion. Check #170 for details.

**Reply by manoelcampos** [reply]
Anyway, this Host and switches bandwidth are very confusing and they don't really work integrated. You can check that in #167.

**Reply by ulfaric** [reply]
I have given up using network cloudlet, because:
1. It only send and receive packets at the beginning, which is not realistic.
2. It requires a target network cloudlet to be there first, casuing trouble in most of cases.

:(

**Reply by manoelcampos** [reply]
1. That depends on your implementation. If you want to send packages during the entire simulation execution, you have to implement that.
2. That is how an actual application works. If you have a client/server app, the client sends packets to the server, which must be running to receive them (or the other way around).

The intent of these Network Cloudlets is to really simulate such a kind of app.

**Reply by manoelcampos** [reply]
Anyway, if you just want to simulate bandwidth consumption without having to create network cloudlets and send packets, maybe just setting a UtilizationModel for the cloudlet BW will do.
